deploy/poseidon-deployment.yaml

```diff
@@ -1,21 +1,130 @@
-apiVersion: v1
-kind: ServiceAccount
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
 metadata:
+  labels:
+    kubernetes.io/bootstrapping: rbac-defaults
+  name: system:poseidon
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:poseidon
+subjects:
+- kind: ServiceAccount
   name: poseidon
   namespace: kube-system
 ---
-kind: Service
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    kubernetes.io/bootstrapping: rbac-defaults
+  name: system:poseidon
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - endpoints
```

Comment:
I am not fully understand why scheduler needs to create endpoints? Is that not one of responsibilities of endpoints controller?

Reply:
We used clusterrole from the default schduler.
You can chek them here, instead of editing the clusterrorle we creaed a new one.

```
kubectl describe clusterrole system:kube-scheduler
```

If you feel something is no needed you can remove them try if it works and raise a PR.

Reply:
Sounds fair, I will try :)